PerfTools/JeProf/src/jeprof.cc

```diff
@@ -0,0 +1,41 @@
+#include "PerfTools/JeProf/interface/jeprof.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <string>
+#include <dlfcn.h>
+#include <cstdio>
+#include <cstring>
+
+extern "C" {
+typedef int (*mallctl_t)(const char *name, void *oldp, size_t *oldlenp, void *newp, size_t newlen);
+}
+
+namespace {
+  bool initialize_prof();
+
+  mallctl_t mallctl = nullptr;
+  const bool have_jemalloc_and_prof = initialize_prof();
+
+  bool initialize_prof() {
+    // check if mallctl and friends are available, if we are using jemalloc
+    mallctl = (mallctl_t)::dlsym(RTLD_DEFAULT, "mallctl");
+    if (mallctl == nullptr)
+      return false;
+    // check if heap profiling available, if --enable-prof was specified at build time
+    bool enable_prof = false;
+    size_t bool_s = sizeof(bool);
+    mallctl("prof.active", &enable_prof, &bool_s, nullptr, 0);
+    return enable_prof;
+  }
+}  // namespace
+
+namespace cms::jeprof {
+  void makeHeapDump(const char *fileName) {
+    if (!have_jemalloc_and_prof)
+      if (!have_jemalloc_and_prof) {
+        edm::LogWarning("JeProfModule") << "JeProfModule requested but application is not"
+                                        << " currently being profiled with jemalloc profiling\n";
```

Comment:
The earlier behavior was to print out the warning only once per job. Printing out on every call might still be ok (there _probably_ aren't that many calls in a program), but I'd nevertheless limit for once per job. A straightforward way would be 
```cpp
#include <mutex>

namespace {
   ...
  std::once_flag warning_flag;
}

...

      if (!have_jemalloc_and_prof) {
        std::call_once(warning_flag, [](){ edm::LogWarning("JeProfModule") ... };
      }
```